habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/InstallDependenciesMojo.java

```diff
@@ -171,23 +171,18 @@ public void doExecute() throws MojoExecutionException, MojoFailureException {
     }
 
     private void setUpInProjectVirtualEnvironment(PoetryCommandHelper poetryHelper) throws MojoExecutionException {
-        String inProjectVirtualEnvironmentPath = HabushuUtil.getInProjectVirtualEnvironmentPath(getPoetryProjectBaseDir());
-        File venv = new File(inProjectVirtualEnvironmentPath);
-        if (this.useInProjectVirtualEnvironment) {
-            if (!venv.exists()) {
-                getLog().info("Configuring Poetry to use an in-project virtual environment...");
-                configureVirtualEnvironmentsInProject(true);
-            }
-        } else {
-            List<String> arguments = new ArrayList<>();
-            arguments.add("config");
-            arguments.add("virtualenvs.in-project");
-            arguments.add("--local");
-            String currentInProjectSetting = poetryHelper.execute(arguments);
-
-            if (Boolean.TRUE.equals(Boolean.valueOf(currentInProjectSetting))) {
-                configureVirtualEnvironmentsInProject(false);
-            }
+        List<String> arguments = new ArrayList<>();
+        arguments.add("config");
+        arguments.add("virtualenvs.in-project");
+        arguments.add("--local");
+        String currentInProjectSetting = poetryHelper.execute(arguments);
+        
+        // update the poetry config to match the useInProjectVirtualEnvironment boolean
+        if (this.useInProjectVirtualEnvironment && Boolean.FALSE.equals(Boolean.valueOf(currentInProjectSetting))) {
+            getLog().info("Configuring Poetry to use an in-project virtual environment...");
+            configureVirtualEnvironmentsInProject(true);
```

Comment:
I: Previously we only updated the poetry config for `virtualenvs.in-project` when a `.venv` didn't exist. This will set the poetry config regardless of if a `.venv` exists, ensuring the `poetry.toml` always matches the value of `useInProjectVirtualEnvironment`.

Reply:
We should test going both in-project versus default location and back for this. I recall that the original logic accounted for a challenge in that sequencing.

Reply:
Tested going back and forth between in project and external virtual environments without any issue. Both the current version (`2.16.1`) and the updated version (`2.17.0-SNAPSHOT`) showed the following behavior:

Case 1: external -> in project
- External environment is deleted
- In project environment is created and used
- `poetry.toml` is updated with `in-project = true`

Case 2: in project -> external
- In project environment is deleted
- External environment is created and used
- `poetry.toml` is updated with `in-project = false`


The only difference in behavior is if you have `habushu.useInProjectVirtualEnvironment=true` and don't have `in-project = true` in your `poetry.toml` but have a `.venv` folder:
- `2.17.0-SNAPSHOT` will add `in-project = true` to your `poetry.toml` and use the in project `.venv`
- `2.16.1` will leave the `poetry.toml` unchanged and create an external environment, ignoring the in project `.venv`

The current behavior of `2.16.1` seems like a bug to me that would be resolved with the `2.17.0-SNAPSHOT` behavior.